src/main/java/com/sforce/ws/types/OffsetDate.java

```diff
@@ -0,0 +1,28 @@
+package com.sforce.ws.types;
+
+import java.time.LocalDate;
+import java.time.ZoneOffset;
+
+public class OffsetDate {
+
+    private final LocalDate date;
+    private final ZoneOffset offset;
+
+    public OffsetDate(LocalDate date, ZoneOffset offset) {
+        this.date = date;
+        this.offset = offset;
+    }
+
+    public LocalDate getDate() {
+        return date;
+    }
+
+    public ZoneOffset getOffset() {
+        return offset;
+    }
+
+    @Override
+    public String toString() {
+        return date.toString() + offset.toString();
+    }
+}
```

Comment:
Why are you creating a new type here? Why don't you use [`java.time.ZonedDateTime`](https://docs.oracle.com/javase/8/docs/api/java/time/ZonedDateTime.html)?

Reply:
Because according to the https://www.w3schools.com/xml/schema_dtypes_date.asp time should be absent, only date is required and timezone is optional for **xs:date**. I have just paste default timezone UTC in case of it's absent in XML, easy to convert to LocalDate/LocalTime/LocalDateTime if needed.

Reply:
Nice. I liked the way implemented the change, without breaking existence runs. I hope salesforce accepts your changes soon.
Sadly, I'm not working on salesforce projects anymore, and I don't have time to setup a test environment and test your changes.